Not working with git confing `worktree.useRelativePaths = true`

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

with the ver of the tool `29.1.0`, git version `2.52.0`, and in git config preset

```
[worktree]
useRelativePaths = true
```

In the [git segment](https://ohmyposh.dev/docs/segments/scm/git) i facing the situation with 0 updates, meaning that it visually looks like it sees that the git repo is in place, but that's it. If I change the absolute paths - everything works.

I changed only the  `.git` file in the worktree, the content

```
gitdir: ../../.git/worktrees/move_to_tf # does not work
gitdir: /workspace/.git/worktrees/move_to_tf # works
```

These paths were generated by `git worktree repair --relative-paths` or `git worktree repair --no-relative-paths.`


### Theme

irrelevant

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

zsh

### Log output

```Shell
[ERROR] 02:53:28.261 terminal.go:RunCommand:277 → exit status 128
[DEBUG] 02:53:28.261 terminal.go:RunCommand:280 → fatal: cannot change to '../../../.worktree/move_to_tf/': No such file or directory
[TRACE] 02:53:28.261 terminal.go:RunCommand(git -C ../../../.worktree/move_to_tf/ --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 1.28525ms
[DEBUG] 02:53:28.261 map.go:String:79 → branch_icon: 


pls note triple `../` instead of double. For worktrees with relative paths, it's the dealbreaker, since the path that is set by git and is untouchable

i tried to debug it, with `sudo ln -s /workspace/.git /.git` meaning that the path is now pointing to the same place, BUT it does not work either without any changes
```

--- a/src/segments/git.go
+++ b/src/segments/git.go
@@ -499,7 +499,7 @@ func (g *Git) hasWorktree(gitdir *runtime.FileInfo) bool {
 		if worktreeIndex > -1 && g.env.HasFilesInDir(g.scmDir, "gitdir") {
 			gitDir := filepath.Join(g.scmDir, "gitdir")
 			realGitFolder := g.env.FileContent(gitDir)
-			g.repoRootDir = strings.TrimSuffix(realGitFolder, ".git\n")
+			g.repoRootDir = strings.TrimSuffix(strings.TrimRight(realGitFolder, "\n\r "), ".git")
 			// resolve relative paths (worktree.useRelativePaths = true)
 			g.repoRootDir = resolveGitPath(g.scmDir, g.repoRootDir)
 			g.scmDir = g.scmDir[:worktreeIndex]
@@ -522,7 +522,7 @@ func (g *Git) hasWorktree(gitdir *runtime.FileInfo) bool {
 		gitDir := filepath.Join(g.mainSCMDir, "gitdir")
 		g.scmDir = g.mainSCMDir[:worktreeIndex]
 		gitDirContent := g.env.FileContent(gitDir)
-		g.repoRootDir = strings.TrimSuffix(gitDirContent, ".git\n")
+		g.repoRootDir = strings.TrimSuffix(strings.TrimRight(gitDirContent, "\n\r "), ".git")
 		// resolve relative paths (worktree.useRelativePaths = true)
 		g.repoRootDir = resolveGitPath(g.mainSCMDir, g.repoRootDir)
 		g.IsWorkTree = true

--- a/src/segments/git_test.go
+++ b/src/segments/git_test.go
@@ -136,6 +136,16 @@ func TestEnabledInWorktree(t *testing.T) {
 			ExpectedRealFolder:    TestRootPath + "dev/worktree",
 			ExpectedRootFolder:    TestRootPath + dotGit,
 		},
+		{
+			Case:                  "worktree with relative gitdir path, no trailing newline",
+			ExpectedEnabled:       true,
+			WorkingFolder:         TestRootPath + "dev/.git/worktrees/folder_worktree",
+			WorkingFolderAddon:    "gitdir",
+			WorkingFolderContent:  "../../../worktree/.git",
+			ExpectedWorkingFolder: TestRootPath + "dev/.git/worktrees/folder_worktree",
+			ExpectedRealFolder:    TestRootPath + "dev/worktree",
+			ExpectedRootFolder:    TestRootPath + dotGit,
+		},
 	}
 	fileInfo := &runtime.FileInfo{
 		Path:         TestRootPath + dotGit,

